lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -447,6 +450,20 @@ class ReportUIFeatures {
         this._toggleDarkTheme();
         break;
       }
+      case 'open-treemap': {
+        // WIP test code :)
+        const treemapData = /** @type {LH.Audit.Details.DebugData} */ (
+          this.json.audits['treemap-data'].details);
+
+        const windowName = `viz-${this.json.requestedUrl}`;
+        const data = {
+          documentUrl: this.json.requestedUrl,
+          id: 'javascript',
+          rootNodes: treemapData.rootNodes,
+        };
+        ReportUIFeatures.openTabAndSendData(data, TREEMAP_URL, windowName);
```

Comment:
Will this always be same origin for treemaps? If so, can we avoid generalizing this and just use (e.g.) indexDB since we're already pulling in a library for it (and caching lhrs from gists in there) anyways? That would allow us to avoid opening up the `targetOrigin` `*` issue even more.

Our xss risk isn't nothing if folks are logged in to github, though the oauth permissions are limited to creating/deleting gists.

Reply:
Only same origin if in viewer. Not the same if the report is a file on disk, or linked to from other integration points (CDT, WPT, PSI).

Reply:
> Only same origin if in viewer. Not the same if the report is a file on disk, or linked to from other integration points (CDT, WPT, PSI).

ok, well, the message event `source` isn't checked, so it needs to be careful it does with the payload.

https://owasp.org/www-community/attacks/xss/ is a helpful starting point.